src/app/cli/src/cli_commands.rs

```diff
@@ -580,6 +583,22 @@ pub fn get_command(
     Ok(command)
 }
 
+pub fn command_needs_transaction(arg_matches: &clap::ArgMatches) -> Result<bool, CLIError> {
+    match arg_matches.subcommand() {
+        Some(("system", system_matches)) => match system_matches.subcommand() {
+            Some(("generate-token", _)) => Ok(true),
+            Some(_) => Ok(false),
+            None => Err(CommandInterpretationFailed.into()),
+        },
+        Some(_) => Ok(false),
+        None => Err(CommandInterpretationFailed.into()),
+    }
+}
```

Comment:
This seems super fragile. What stops us from always having some transaction runner in the catalog (real DB or noop impl) and always run commands in transactional scope?

Reply:
we have commands that take a long time to execute -- e.g.  API server

can't do that because the commit will never happen

-- 

The key point here is not the presence of the component itself in the Catalog (they are already present, after all), but the launch of the transaction

Reply:
and yes, I was bitterly disappointed to discover that we need to know if a transaction is required BEFORE creating a Command